src/test/test_proj_data_info_subsets.cxx

```diff
@@ -0,0 +1,332 @@
+      everything_ok = false;
+  }
+}
+
+
+void TestProjDataInfoSubsets::
+test_split(const ProjData &proj_data)
+{
+  cerr << "\tTesting ability to split a ProjData into consistent subsets" << endl;
+  int view;    
+  int num_subsets = 4;
+
+  std::vector<int> subset_views;
+
+  for (int subset = 0; subset < num_subsets; ++subset){
+    // create a vector containg every num_subsest-th view starting at subset
+    // for num_subsets = 4 and subset = 0 this is [0, 4, 8, 12, ...]
+    // for num_subsets = 4 and subset = 1 this is [1, 5, 9, 13, ...]
+
+    subset_views.clear();
+    view = subset;
+
+    while(view < proj_data.get_num_views()){
+      subset_views.push_back(view);
+      view += num_subsets;
+    }
+
+    auto subset_proj_data_uptr = proj_data.get_subset(subset_views);
+    auto& subset_proj_data = *subset_proj_data_uptr;
+
+    // loop over views in the subset data and compare them against the original "full" data
+    for(std::size_t i = 0; i < subset_views.size(); ++i)
+      {
+        // i runs from 0, 1, ... views_in_subset - 1 and indicates the view number in the subset
+        // the corresponding view in the original data is at subset_views[i]
+
+        // loop over all segments to check viewgram for all segments
+        for (int segment_num = proj_data.get_min_segment_num(); segment_num < proj_data.get_max_segment_num(); ++segment_num)
+          {
+            check_if_equal(
+                           proj_data.get_viewgram(subset_views[i], segment_num),
+                           subset_proj_data.get_viewgram(i, segment_num), "Are viewgrams equal?");
+            // TODO also compare viewgram metadata
+          }
+      }
+  }      
+}
+
+
+// void TestProjDataInfoSubsets::
+// test_split_and_combine(const ProjData &proj_data, int num_subsets)
+// {
+//     StandardSubsetter subsetter = StandardSubsetter(proj_data.get_proj_data_info_sptr(), num_subsets);
+
+//     std::vector<ProjData> subsets;
+//     for (int s=0; s++; s<num_subsets) {
+//         //ProjData& subset = *proj_data.get_subset(s, num_subsets);
+//         // or
+//         ProjData& subset = *proj_data.get_subset(subsetter.get_views_for_subset(s));
+//         subsets.push_back(subset);
+//     }
+
+//     ProjData new_proj_data = ProjData(proj_data);  // how to copy?
+//     new_proj_data.fill(0);
+
+//     for (int s=0; s++; s<num_subsets) {
+//         new_proj_data.fill_subset(s, num_subsets, subsets[s]);
+//     }
+
+//     compare_sinos(proj_data, new_proj_data);
+// }
+
+
+void TestProjDataInfoSubsets::
+test_forward_projection_is_consistent(
+    const shared_ptr<const VoxelsOnCartesianGrid<float> >& input_image_sptr,
+    const shared_ptr<const ProjData>& template_sino_sptr,
+    const shared_ptr<ForwardProjectorByBin>& fwd_projector_sptr,
+    bool use_symmetries, int num_subsets)
+{
+  cerr << "\tTesting Subset forward projection is consistent" << endl;
+  
+  auto full_forward_projection = ProjDataInMemory(*template_sino_sptr);
```

Comment:
while this compiles, it means that `full_forward_projection` will be a reference to an object that is immediately deleted.

Reply:
This is what I did trying to localize the source of segfault.

Firstly, I replaced 247 with
```
  ProjDataInMemory full_forward_projection(*template_sino_sptr);
```
(I actually believe clever enough compiler does that anyway, but just in case).
Next I put `return` after line 269 - still segfalts, so segfault does not seem to be caused by 270 (or at least not only by 270, which causes that weird alleged 'different geometry' error we observe).
I then put
```
if (use_symmetries)
```
before 127 so that it does not affect full projection - no segfault!

So, it looks like the trouble is caused by `proj_pair_sptr->set_up`.